eventloop/eventloop.go

```diff
@@ -199,25 +199,29 @@ func (loop *EventLoop) Start() {
 // Calling Stop() on a non-running loop has no effect.
 // It is not allowed to call Stop() from the loop, because it is synchronous and cannot complete until the loop
 // is not running any jobs. Use StopNoWait() instead.
-func (loop *EventLoop) Stop() {
+// return number of jobs remaining
+func (loop *EventLoop) Stop() int32 {
 	loop.stopLock.Lock()
 	for loop.running {
 		atomic.StoreInt32(&loop.canRun, 0)
 		loop.wakeup()
 		loop.stopCond.Wait()
 	}
 	loop.stopLock.Unlock()
+	return loop.jobCount
 }
 
 // StopNoWait tells the loop to stop and returns immediately. Can be used inside the loop. Calling it on a
 // non-running loop has no effect.
-func (loop *EventLoop) StopNoWait() {
+// return number of jobs remaining
+func (loop *EventLoop) StopNoWait() int32 {
```

Comment:
Unfortunately it won't work for StopNoWait(), because at the time when loop.jobCount is accessed the loop may still be running thus the access will create a data race.